### DISCUSSION TITLE
Reset index from within app

### DISCUSSION THREAD

**Discussion by xiwenc** [MobiusSync/MobiusSync#176] [Q&A] [answered]
Hi, thanks for creating Mobius!

Sometimes my share is stuck with out-of-sync items. When there are a few files i can deal with them manually.

However sometimes there are hundreds. This seems to be a known issue with syncthing in general. On my computers i could resetting indexes by calling a REST api. However that is not possible on iOS. Or at least i dont know how to.

Is it possible to add some UI in the app that will make this REST call? Background info can be found here: https://forum.syncthing.net/t/out-of-sync-items-can-only-ever-fix-it-by-removing-the-index-v0-14-0-db/18240 

curl -X POST -H "X-API-Key: dHM5aXsqPSKahk7NninFf3" http://localhost:8384/rest/system/reset?folder=uq9xm-mfn
{"ok": "resetting folder uq9xm-mfn"}

Ideally this can be added as a button "reset" next to "rescan" inside of the folder section.

If that's too much effort, it could also work with a simple prompt that asks for folder id. Similar approach of the purchase menu.

Thanks in advance!

**Comment by xiwenc** [comment]
![image](https://github.com/MobiusSync/MobiusSync/assets/1109695/b5cb9924-2bb6-4f7c-9376-7e8c88a93a77)

Tried to create a shortcut that calls this API. However it doesnt seem to work. Perhaps the syncthing process runs inside some kind of container/sandbox?

**Comment by SimonPickup** [comment, marked as answer]
Correct, the API is not exposed outside the app sandbox.

You should be able to remove the folder in Möbius Sync then re-add it without having to copy all the files again.

**Comment by xiwenc** [comment]
Thanks for the reply!

Makes sense. Thanks!
